index.js

```diff
@@ -61,7 +61,10 @@ RedirectableRequest.prototype.write = function (data, encoding, callback) {
   if (!(typeof data === "string" || typeof data === "object" && ("length" in data))) {
     throw new Error("data should be a string, Buffer or Uint8Array");
   }
-  if (this._requestBodyLength + data.length <= this._options.maxBodyLength) {
+  if(data instanceof Buffer && data.length === 0) {
+    callback();
```

Comment:
While this seems to fix the problem, I'm still curious as to why the problem occurs. We are passing `data` to `this._currentRequest.write` below. Why doesn't that call the callback (in the case where the buffer is empty)?